core/src/program/blocks/dyn_block.rs

```diff
@@ -0,0 +1,56 @@
+use super::{fmt, hasher, Digest, Felt, Operation};
+
+// Dyn BLOCK
+// ================================================================================================
+/// Block for a dynamic function where the target is specified by the stack.
+///
+/// Executes the function referenced by the hash on top of the stack. Fails if the body is
+/// unavailable to the VM, or if the execution of the dynamically-specified function fails.
+///
+/// The hash of a dyn block is computed as:
+///
+/// > hash(DYN_CONSTANT || padding, domain=CALL_DOMAIN)
+///
+/// Where `fn_hash` is 4 field elements (256 bits), and `padding` is 4 ZERO elements (256 bits).
+/// TODO: check on this hashing. Does it make sense?
+#[derive(Clone, Debug)]
+pub struct Dyn {
```

Comment:
Yeah, I wasn't sure if we were going to switch to the `Digest::default()` method, but sticking with this keeps things more consistent with the other code blocks. Since we're keeping it, I'll update these and also add a pre-computed constant. No sense doing this work unnecessarily